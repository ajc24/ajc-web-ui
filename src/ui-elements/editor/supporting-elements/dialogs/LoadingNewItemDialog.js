import React from 'react';
import PropTypes from 'prop-types';
import DialogWithPageMask from '../../../dialog/DialogWithPageMask';

/**
 * The LoadingNewItemDialog component renders a dialog for when the application is in the process of adding a new item to the editor
 */
class LoadingNewItemDialog extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <DialogWithPageMask id="loading-new-item-dialog-id" dialogContentAreaColour={this.props.dialogContentAreaColour} colour={this.props.colour}
        dialogTitle="Please Wait... Adding Your New Item" buttonsList={[]} isDisplayed={this.props.isDisplayed} onClose={undefined}>
      </DialogWithPageMask>
    );
  }
}
LoadingNewItemDialog.propTypes = {
  /** The colour / theme set to the dialogs panels. */
  colour: PropTypes.oneOf([ 'grey', 'red' ]),
  /** The background colour of the dialogs content area. */
  dialogContentAreaColour: PropTypes.oneOf([ 'grey', 'white', 'yellow' ]),
  /** Determines whether the dialog is displayed or not. */
  isDisplayed: PropTypes.bool,
};
LoadingNewItemDialog.defaultProps = {
  colour: 'grey',
  dialogContentAreaColour: 'white',
  isDisplayed: false,
};
export default LoadingNewItemDialog;
