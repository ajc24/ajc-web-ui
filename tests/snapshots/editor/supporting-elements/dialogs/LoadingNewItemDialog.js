import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import LoadingNewItemDialog from '../../../../../src/ui-elements/editor/supporting-elements/dialogs/LoadingNewItemDialog';

describe('LoadingNewItemDialog', () => {
  describe('Default props and rendering', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <LoadingNewItemDialog />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - All properties custom set', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <LoadingNewItemDialog colour="red" dialogContentAreaColour="grey" isDisplayed={true} />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
