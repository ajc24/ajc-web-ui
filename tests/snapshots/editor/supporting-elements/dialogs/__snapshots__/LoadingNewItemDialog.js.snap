// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingNewItemDialog Default props and rendering verifies the snapshot for the component 1`] = `null`;

exports[`LoadingNewItemDialog Transferred props and rendering - All properties custom set verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-center ajc-page-mask ajc-width-4k"
  id="page-mask-loading-new-item-dialog-id-root"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-center ajc-width-hd"
    >
      <div
        className="ajc-flex-column ajc-flex-justify-start ajc-spacing-top-large ajc-spacing-bottom-large ajc-width-hd-with-padding ajc-text-white ajc-font-default"
        id="page-mask-loading-new-item-dialog-id-content"
      >
        <div
          className="ajc-flex-row ajc-flex-justify-center ajc-width-hd-inner"
        >
          <div
            className="ajc-flex-column ajc-flex-justify-left ajc-width-dialog-parent"
            id="loading-new-item-dialog-id-root"
          >
            <div
              className="ajc-flex-row ajc-flex-justify-left ajc-dialog-title ajc-width-dialog ajc-text-white ajc-font-default ajc-background-red"
              id="loading-new-item-dialog-id-title"
            >
              <h2
                className="ajc-font-dialog-title"
              >
                Adding Your New Item
              </h2>
            </div>
            <div
              className="ajc-flex-row ajc-flex-justify-left ajc-dialog-content ajc-width-dialog-content ajc-background-grey-1"
              id="loading-new-item-dialog-id-content-area-root"
            >
              <div
                className="ajc-flex-column ajc-flex-justify-left ajc-text-black ajc-font-default"
                id="loading-new-item-dialog-id-content-area"
              >
                <p
                  className="ajc-flex-row ajc-background-transparent ajc-page-content-text ajc-spacing-bottom-small ajc-text-black ajc-font-default ajc-flex-justify-left"
                >
                  Please wait just a moment. We are getting your new item ready for you.
                </p>
              </div>
            </div>
            <div
              className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-red"
              id="loading-new-item-dialog-id-footer"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
